### PR TITLE
Add support for speech recognition infer only without eval

### LINKED ISSUE #2618: How can I inference just one audio by Pre-train Wav2vec 2.0 model?
How can I inference just one audio by Pre-train Wav2vec 2.0 model?  
`examples/speech_recognition/infer.py` too difficult to me.

### LINKED ISSUE #2651: wav2vec 2.0 inference pipeline
## 🚀 Feature Request
Provide a simple inference pipeline for the `wav2vec 2.0` model.

### Motivation
Current inference script `examples/speech_recognition/infer.py` handles a lot of cases, resulting being extremely complex.
### Pitch
A single python script that loads and runs inference with `wav2vec 2.0` pre-trained model on a single wav file or on a programmatically loaded waveform signal.

### Alternatives

-
### Additional context
This kind of inference pipeline would enable indi researchers to test the model on their audio dataset and and against other models.

### PR DESCRIPTION
**Pull request by lyogavin**
# Before submitting

- [X] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [X] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/master/CONTRIBUTING.md)?
- [X] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Fixes #2651 
Fixes #2618 

Often we want to only speech recognize some audio files, we don't have the labels or don't want to construct the data set with labels.

This PR support "infer" only run.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by tensorfoo** [comment]
is this documented anywhere? how are you supposed to use it?
